docs/DeveloperGuide.adoc

```diff
@@ -364,6 +365,136 @@ We are using `java.util.logging` package for logging. The `LogsCenter` class is
+method in the `UserPrefs` class using Java Reflection API. If an exception is encountered, it is assumed that the method
+does not exist and the preference key entered by the user was invalid. Similarly, if two arguments were entered, `PrefCommand`
+will again try to find and execute `set<PreferenceKey>` method.
+
+==== Design Considerations
+
+**Aspect:** Implementation of `PrefCommand` +
+**Alternative 1 (current choice):** Use Reflection API +
+**Pros:** Easier to add in a new preference by changing only the `UserPrefs` class +
+**Cons:** Hard to debug since it is less intuitive +
+**Alternative 2:** Design another parser similar to the command parser +
+**Pros:** Easier to understand the code +
+**Cons:** More difficult to add in a new preference
+
+---
+
+**Aspect:** Data structure to store the preferences : +
+**Alternative 1 (current choice):** JSON and jackson-databind +
+**Pros:** Most of the parsing is taken care of by the API +
+**Cons:** Adding new preference is slightly troublesome; the API documentation is not beginner-friendly +
+**Alternative 2:** Use a hashmap to store the preferences +
+**Pros:** Intuitive, easy to add more preferences +
+**Cons:** Data structure is more restricted; may need to write native code to parse JSON if a library cannot be found to convert JSON to hashmap +
+// end::preferences[]
+
+==== Command: `List By Most Searched`
+
+Each successful `Find` command execution with non-zero search results, activates an update of search count for everyone in search results.+
+[source,java]
+----
+if (searchResultsCount != NO_RESULTS) {
+    model.recordSearchHistory();
+}
+
+----
+This equates to an update of the search count variable within the `SearchData` class within every `Person` class.
+[source,java]
+----
+public void incrementSearchCount()
+----
+
+When the `ListByMostSearched` is executed, `UniquePersonList` is sorted based on descending search count. +
+[source,java]
+----
+public void sortBySearchCount () {
+    internalList.sort(new ReadOnlyPersonComparator());
+}
+----
+The `ReadOnlyPersonComparator` compares the search count between two Persons. This updates the address book storage file.+
+When model calls for `filteredPersons` list, all persons are read from +
+the address book storage and then displayed on `PersonListPanel` on the GUI.
+
+
+=== Design Considerations
+
+**Aspect:** Implementation of storage for `SearchData` +
+**Alternative 1 (current choice):** Add `SearchData` as a class in `Person` +
+**Pros:** Increase storage efficiency +
+**Cons:** `Addressbook.xml` is more complicated
+**Alternative 2:** Add `SearchData` to a separate storage file +
+**Pros:** Minimal changes to `Addressbook.xml` +
+**Cons:** Unnecessary repetition in storage; replication of `Person` details +
+
+
+---
+
+**Aspect:** Persistent sort +
+**Alternative 1(current choice):** Persistent sort (Sort persons, by modifying `addressbook.xml`) +
+**Pros:** Reduces need for repeated sorting & speed up `find` command +
```

Comment:
Extract from the implementation section for `undo/redo`.

```
It supports undoing and redoing of commands that modifies the state of the address book 
```

So since your `lms` command does modify the state of the addressbook, you should either support undo-ing/redo-ing a sort, or make a special note in the undo/redo section.